### PR TITLE
Update store implementation

### PR DESCRIPTION
**Pull request by forman**
Adapted `SentinelHubStore` to conform to proposed changes in https://github.com/dcs4cop/xcube/pull/359

### CODE REVIEW

File: xcube_sh/store.py [hunk 1]
```diff
@@ -306,30 +306,33 @@ def get_data_store_params_schema(cls) -> JsonObjectSchema:
...
-        return TYPE_ID_DATASET,
+    def get_type_specifiers(cls) -> Tuple[str, ...]:
+        return str(TYPE_SPECIFIER_CUBE),
 
-    def get_data_ids(self, type_id: str = None) -> Iterator[Tuple[str, Optional[str]]]:
-        self._assert_valid_type_id(type_id)
-        metadata = SentinelHubMetadata()
-        for data_id, dataset in metadata.datasets.items():
-            yield data_id, dataset.get('title')
+    def get_type_specifiers_for_data(self, data_id: str) -> Tuple[str, ...]:
```
Comment (pont-us) [inline xcube_sh/store.py:312]: Should this method check that the supplied `data_id` is valid?

File: xcube_sh/store.py [hunk 2]
```diff
@@ -306,30 +306,33 @@ def get_data_store_params_schema(cls) -> JsonObjectSchema:
...
+        return self.get_type_specifiers()
 
-    def has_data(self, data_id: str) -> bool:
+    def get_data_ids(self, type_specifier: str = None, include_titles=True) -> Iterator[Tuple[str, Optional[str]]]:
+        if self._is_supported_type_specifier(type_specifier):
+            metadata = SentinelHubMetadata()
+            for data_id, dataset in metadata.datasets.items():
+                yield data_id, dataset.get('title')
```
Comment (pont-us) [inline xcube_sh/store.py:319]:
```suggestion
                yield data_id, (dataset.get('title') if include_titles else None)
```

The docstring for the `get_data_ids` abstract method says "If *include_titles* is false, the second item of the result tuple will be None.".

File: xcube_sh/store.py [hunk 3]
```diff
@@ -306,30 +306,33 @@ def get_data_store_params_schema(cls) -> JsonObjectSchema:
...
+
+    def has_data(self, data_id: str, type_specifier: str = None) -> bool:
```
Comment (pont-us) [inline xcube_sh/store.py:321]: Type specifier should be checked.

**Review by pont-us — CHANGES_REQUESTED** [review]

File: test/test_store.py [hunk 4]
```diff
@@ -65,3 +65,14 @@ def test_data_opener_params_schema(self):
         self.assertIn('spatial_res', schema.properties)
         self.assertIn('bbox', schema.properties)
         self.assertIn('crs', schema.properties)
+
+    def test_get_type_specifiers(self):
```
Comment (TonioF) [inline test/test_store.py:69]: Please add @unittest.skipUnless(HAS_SH_CREDENTIALS, REQUIRE_SH_CREDENTIALS) to both tests. Also, I found that test_data_opener_params_schema fails when the credentials are not set, as more parameters are required.

**Review by TonioF — CHANGES_REQUESTED** [review]
Just comments on tests, store changes look fine!

File: xcube_sh/store.py (hunk 3 above)
Comment (forman) [reply xcube_sh/store.py:321]: Yes.

File: xcube_sh/store.py (hunk 1 above)
Comment (forman) [reply xcube_sh/store.py:312]: Yes.

File: test/test_store.py (hunk 4 above)
Comment (forman) [reply test/test_store.py:69]: Right.

**Pull request merged**